### PR TITLE
Kill off OMF support

### PR DESCRIPTION
**Pull request by rikkimax**
This is being discussed at BeerConf that it is time to go.

DMC is dead (that I remember being said by @WalterBright quite a while back).

So this removes the switch.

see also https://github.com/dlang/dmd/pull/13110 for the deprecation of this switch

### CODE REVIEW

**Comment by thewilsonator** [comment]
Will also need to kill off the CI: `.azure-pipelines/windows.sh` (some hits for "omf") is a start

**Comment by rikkimax** [comment]
Nic has been extremely keen to see OMF removed today, he has been doing investigative work point out to me what needs changing to rip out the ability for people to use it.

**Comment by rikkimax** [comment]
CI is green!

Oh, and @LightBender says he wants to take you out for some after-funeral drinks if you're up for it!

**Review by RazvanN7 — APPROVED** [review]

**Review by thewilsonator — APPROVED** [review]

File: compiler/src/dmd/mars.d [hunk 1]
```diff
@@ -901,11 +901,6 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
             target.isX86_64 = false;
             target.omfobj = false;
         }
-        else if (arg == "-m32omf") // https://dlang.org/dmd.html#switch-m32omfobj
```
Comment (dkorpel) [inline compiler/src/dmd/mars.d:904]: As long as we keep the code generation logic, perhaps this could stay as an undocumented switch?

Comment (thewilsonator) [reply compiler/src/dmd/mars.d:904]: The idea is to kill off the rest of it once this pulled in stages.

Comment (rikkimax) [reply compiler/src/dmd/mars.d:904]: We still have to deal with the installer so a release can happen.

Cleaning the rest of the OMF support can wait, it's going to be a bit to untangle I expect.

**Comment by ibuclaw** [comment]
This caused a regression.

```
Building Druntime 32omf
Entering dir: clones\dmd\druntime
Running: mingw32-make -j4 MODEL=32omf "DMD=C:\Users\vagrant\clones/dmd/generated/windows/release/32/dmd.exe" ENABLE_RELEASE=1 LATEST=2.109.0-beta.1
Using make SHELL "C:/Program Files/Git/usr/bin/bash.exe", should be bash.
C:\Users\vagrant\clones/dmd/generated/windows/release/32/dmd.exe -c -conf= -Isrc -Iimport -w -de -preview=dip1000 -preview=fieldwise -m32omf   -preview=dtorfields -O -release -inline  -I. -P=-I. src/core/stdc/errno.c -of../generated/windows/release/32omf/errno_c.obj
Error: unrecognized switch '-m32omf'
       run `dmd` to print the compiler manual
       run `dmd -man` to open browser on manual
mingw32-make: *** [Makefile:391: ../generated/windows/release/32omf/errno_c.obj] Error 1
```

**Comment by ibuclaw** [comment]
Druntime appears forgotten.

https://github.com/dlang/dmd/blob/master/druntime/Makefile#L378

**Comment by ibuclaw** [comment]
And likely phobos too.

https://github.com/dlang/phobos/blob/master/Makefile#L199

**Comment by ibuclaw** [comment]
> Dub, druntine, and Phobos like the internals can be cleaned up. The installer however I thought Nic was on that. Is that the source of the error you posted?

There's been no activity for over 2 weeks on the installer repo.

The sequencing of this seems backwards though. First drop it from the toolchain, then remove it from the compiler.

**Pull request merged**